codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -215,6 +247,25 @@ let updateInternal = (
     )
   }
 
+  let latestFullyFetchedBlock = latestFullyFetchedBlock.contents
+
+  let isFetchingAtHead = switch currentBlockHeight {
+  | None => fetchState.isFetchingAtHead
+  | Some(headBlockNumber) =>
```

Comment:
The only thing that, compared to before, this logic will apply to every response instead of only dc registration.

But this shouldn't break anything, because there's usually > 1_000_000 blocks on a chain, and this is already a 1_000 blocks sync range.